### PR TITLE
Influxless tests (closes #18)

### LINKED ISSUE #18: Reduce/remove testing dependency on historical data
Right now the main functionality test for moneybot relies on a fully-populated historical dataset. At 163 MB this is not a trivial thing to throw around and having to load it before test runs is going to make #14 very difficult. So, here are two proposals for how to solve this, in order of awesomeness:

1. Completely mock out calls to InfluxDB during testing. This would be my preference as it completely frees the testing setup of external dependencies and will make CI much, much simpler. We would manually craft the responses we expect to get for a given set of queries and mock out the appropriate calls to return these responses. This lets us test what we actually care about: whether `moneybot` does the right thing with the data it gets from InfluxDB (as opposed to whether we can set up InfluxDB the right way).

2. Create a much smaller dataset that can be used for testing. This is less ideal, as it still requires us to make an instance of InfluxDB available to the tests.

Let me know what you guys think. I'm happy to handle this one, I just don't know what kind of data we should be expecting to get from InfluxDB.

### PR DESCRIPTION
**Pull request by elsehow**
This PR removes our dependence on a seeded (and running) InfluxDB instance during tests.

Specifically, this PR makes a mock of the `MarketHistory` class. Why mock this class, instead of the InfluxDB client? Well, notice that `client`s are only used in two places: the `LiveMarketAdapter` and `MarketHistory`. These are messy, side-effecty classes that do things like scrape external APIs and so on. By making `MarketHistoryMock` only implement only a few, select methods, we make explicit that we are *not* mocking the other things that `MarketHistory` does, like scraping. Someday, we may want to mock functionality like this (e.g. #13), but since our current `test_basic.py` doesn't call for such tests, it makes sense to keep the mocks as parsimonious as possible w/r/t the test case.

### CODE REVIEW

**Comment by elsehow** [comment]
All sounds good. Newest commit trades pickles for json and starts `moneybot.testing`.

File: moneybot/testing.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from datetime import datetime
+import json
+from typing import Dict, List
+import pandas as pd
+
+
+class MarketHistoryMock:
+
+    _history = json.load(open('tests/mock-data/history.json', 'r'))
+    _charts = json.load(open('tests/mock-data/charts.json', 'r'))
```
Comment (adamrothman) [inline moneybot/testing.py]: Doing it this way, I'm not sure the file handles will ever get closed. How about initializing these to `None`, then loading them using `with open(...) as ...` in `__init__` (iff they haven't been already)?

Comment (adamrothman) [inline moneybot/testing.py]: `type(self)` is the conventional way to get a reference to the class object from an instance, e.g. `type(self)._history`.

Comment (elsehow) [inline moneybot/testing.py]: Interesting - why will this assure the files get closed?

Comment (adamrothman) [inline moneybot/testing.py]:
```python
# Done this way, f must be closed explicitly
f = open('/path/to/file.json')
stuff = json.load(f)
f.close()

# But file handles are also context managers
with open('/path/to/file') as f:
    # f is automatically closed when exiting scope (even if an exception is raised)
    stuff = json.load(f)
```

Comment (elsehow) [inline moneybot/testing.py]: That makes some sense. It's like the original file object is getting garbage collected.

File: moneybot/testing.py [hunk 2]
```diff
@@ -6,8 +6,11 @@
 
 class MarketHistoryMock:
 
-    _history = json.load(open('tests/mock-data/history.json', 'r'))
-    _charts = json.load(open('tests/mock-data/charts.json', 'r'))
+    def __init__(self):
+        with open('tests/mock-data/history.json', 'r') as f:
+            self._history = json.load(f)
+        with open('tests/mock-data/charts.json', 'r') as f:
+            self._charts = json.load(f)
```
Comment (adamrothman) [inline moneybot/testing.py]: Sorry, not to nit-pick but –

As implemented, these JSON files will be reloaded every time a new instance of `MarketHistoryMock` is created. Not a big deal now as we only have one test that uses the class, but not something we want to worry about in the future.

```python
class MarketHistoryMock:

    _history = None
    _charts = None

    def __init__(self):
        cls = type(self)
        if cls._history is None:
            # Files are opened in 'r' mode by default
            with open('tests/mock-data/history.json') as f:
                cls._history = json.load(f)
        if cls._charts is None:
            with open('tests/mock-data/charts.json') as f:
                self._charts = json.load(f)
```

This reads the files at most once, when the first instance is created. By making `_history` and `_charts` class variables, every instance can share the same data.

Comment (elsehow) [inline moneybot/testing.py]: mypy compains about them being set to `None`

```
moneybot/testing.py:22: error: Value of type None is not indexable
moneybot/testing.py:32: error: Value of type None is not indexable
```
And, when I set them to empty dicts, I get a key error...

Comment (adamrothman) [inline moneybot/testing.py]: Ah right, because it can't infer the type. Two options:

1. Add [inline annotations](http://mypy.readthedocs.io/en/latest/type_inference_and_annotations.html#explicit-types-for-variables): `_history = None  # type: Dict`
2. Initialize them as empty dicts and change the check from `if cls._history is None` to `if len(cls._history) == 0`.

IMO the first is superior.

Comment (elsehow) [inline moneybot/testing.py]: python magic

**Pull request merged**